hooks/VDM.js

```diff
@@ -268,19 +304,304 @@ module.exports = function (input, session) {
+      })
+    }
+  }
+
+  if (data.waveperiod) {
+    contextPrefix = 'meteo.'
+    if (data.waveperiod < 61) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.wave.period`,
+        value: data.waveperiod,
+      })
+    }
+  }
+
+  if (data.wavedir) {
+    contextPrefix = 'meteo.'
+    if (data.wavedir < 360) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.wave.direction`,
+        value: utils.transform(data.wavedir, 'deg', 'rad'),
+      })
+    }
+  }
+
+  if (data.swellhgt) {
+    contextPrefix = 'meteo.'
+    if (data.swellhgt < 252) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.swell.height`,
+        value: data.swellhgt / 10,
+      })
+    }
+  }
+
+  if (data.swellperiod) {
+    contextPrefix = 'meteo.'
+    if (data.swellperiod < 61) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.swell.period`,
+        value: data.swellperiod,
+      })
+    }
+  }
+
+  if (data.swelldir) {
+    contextPrefix = 'meteo.'
+    if (data.swelldir < 360) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.swell.direction`,
+        value: utils.transform(data.swelldir, 'deg', 'rad'),
+      })
+    }
+  }
+
+  if (data.seastate) {
+    contextPrefix = 'meteo.'
+    if (data.seastate < 13) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.seaState`,
+        value: beaufortScale[data.seastate],
+      })
+    }
+  }
+
+  if (data.watertemp) {
+    contextPrefix = 'meteo.'
+    if (data.watertemp > -101 && data.watertemp < 501) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.water.temperature`,
+        value: utils.transform((data.watertemp / 10), 'c', 'k'),
+      })
+    }
+  }
+
+  if (data.precipitation) {
+    contextPrefix = 'meteo.'
+    if (data.precipitation < 7) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.precipitation`,
+        value: precipitationType[data.precipitation],
+      })
+    }
+  }
+
+  if (data.salinity) {
+    contextPrefix = 'meteo.'
+    if (data.salinity < 502) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.water.salinity`,
+        value: data.salinity / 10,
+      })
+    }
+  }
+
+  if (data.ice) {
+    contextPrefix = 'meteo.'
+    if (data.ice < 3) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.ice`,
+        value: iceTable[data.ice],
+      })
+    }
+  }
+
   if (values.length === 0) {
     return null
   }
 
-  const delta = {
-    context: contextPrefix + `urn:mrn:imo:mmsi:${data.mmsi}`,
-    updates: [
-      {
-        source: tags.source,
-        timestamp: tags.timestamp,
-        values: values,
-      },
-    ],
+  if (contextPrefix === 'meteo.'){
+    if (data.lon && data.lat) {
+      values.push({
+        path: `environment.observations.${meteoLocation}`,
```

Comment:
I'm not a fan of `metoLocation` in the path as it is not consistent with other contexts.
I would prefer to see this align witrh SK convention and use:
context = `meteo.{identifier}`
path = `environment.observations.xxx`
Weather station Positon = `navigation.position`

Reply:
I'm neither, but as there are data sources with same mssi but different locations, then this was one proposal to handle these.

If weather stations would respect even some rules, for mssi, then there would not be any issues at all.

Reply:
Can there be more than one weather station associated with an MMSI?